packages/stateManager/mod.ts

```diff
@@ -92,6 +95,28 @@ async function storeOrdersByStatus(
   }
   return store.put(status, orders);
 }
+// Synchronize db operations to prevent race conditions.
+const lockAndRelease = <T extends ShopObjectTypes>(
+  store: Store<T>,
+  dbMutex: Mutex,
+): Store<T> => {
+  return {
+    put: (key: StoreKey, value: T) => {
+      return dbMutex.runExclusive(() => {
+        debug(`DB Write: ${String(key)}`);
+        return store.put(key, value);
+      });
+    },
+    get: (key: StoreKey) => {
+      return dbMutex.runExclusive(() => {
+        return store.get(key);
+      });
+    },
+    iterator: () => {
+      return store.iterator();
```

Comment:
why don't we lock this iterator?

we shouldn't read while writes are happening.